### PR TITLE
generic task

### PR DESCRIPTION
**Pull request by spacex33**

### CODE REVIEW

File: src/SomeClass.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import java.util.Arrays;
+import java.util.List;
+
+public class SomeClass {
+
+    public static <T extends Number> void evenNumberCount(List<T> numbers){
+        int evenNumbers = 0;
+        for (T num : numbers) {
+            if (num instanceof Integer && num.intValue() % 2 == 0){
```
Comment (stefankiv) [inline src/SomeClass.java]: Almost. What if I pass list of doubles to the method?
`SomeClass.evenNumberCount(Arrays.asList(2.0, 3.0, 4.0))`
I'd expect to see `2` as a result.

File: src/SomeClass.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import java.util.Arrays;
+import java.util.List;
+
+public class SomeClass {
```
Comment (stefankiv) [inline src/SomeClass.java]: It isn't related to generics, but if you are designing methods which do something, it's useful to return a result of the method execution, if possible. I mean if you count even numbers then return that count. And so on :)

File: src/SomeClass.java (hunk 1 above)
Comment (spacex33) [reply src/SomeClass.java]: I found my mistake and switched code to " if (num.doubleValue() % 2 == 0) ", according to my few tests, now it works correctly. 
Should I make a new commit with changes?

File: src/SomeClass.java (hunk 2 above)
Comment (spacex33) [reply src/SomeClass.java]: I thought that if the return statement is really necessary, you would mentioned this in the task, so I allowed myself to do it in another way
but I understand, that return it's useful, so in the future I will use it by default

File: src/SomeClass.java (hunk 1 above)
Comment (stefankiv) [reply src/SomeClass.java]: Yep, why not? :)

File: src/SomeClass.java (hunk 2 above)
Comment (stefankiv) [reply src/SomeClass.java]: It's okay, I just wanted to be sure you understand it.

**Pull request closed without merging**